### DISCUSSION TITLE
[@types/passport-jwt]: Making the library Express agnostic creates issues with the other `passport` type libraries, which are not Express agnostic :D

### DISCUSSION THREAD

**Discussion by ValTashkov** [DefinitelyTyped/DefinitelyTyped#68174] [Issues with a @types package] [answered]
With the newest version (3.4) of `@types/passport-jwt` (after this commit-> https://github.com/DefinitelyTyped/DefinitelyTyped/commit/79fd6e07f7c065d0de3f2baffac1006ad7292da3) the types don't work properly with Express Request type which is required for `passport-strategy` 

To be a bit more precise
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/137040855/ef0a3b49-ba82-438a-80ee-4b6a4a49a519)
The line seems great. We add a Generic that defaults to the global Request and should make it framework agnostic

BUT

Here all these functions that return  `JwtFromRequestFunction` don't set a Generic type to it and we don't have a way to pass one
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/137040855/391f88b1-abad-4552-a007-3a87965ffc30)
Which in turn conflicts with `passport-strategy` which has `Strategy.authenticate` that requires Express Request:
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/137040855/261597c7-bc04-41f7-bb77-cc7a5adaa611)
And, again, currently we don't set a different Generic type to `JwtFromRequestFunction` ( I opened a PR for that below )

Also I noticed that `Express.User` is still used:
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/137040855/414d2190-b424-4219-980b-893e2e161011)
Which comes from `Express` defined in `passport`. 

So that begs the question:
Why make one of the passport libraries Express agnostic when all the others are not ?
Shouldn't change of that scale be made to all related libraries ?

[a comment by an automated account is not shown]

**Comment by ValTashkov** [comment]
I have opened a PR with a suggested change -> https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68173

**Comment by ValTashkov** [comment]
Also I will tag @QuentinLemCode as he is the person that made the change

**Comment by QuentinLemCode** [comment, marked as answer]
Hello @ValTashkov 
In fact, the generic Request object wasn't a good choice. I have opened another PR to fix this : #68102 
Can you test with the code from this PR please ?

**Reply by ValTashkov** [reply]
Yep, that seems to fix the issue. I'll close my PR. Thanks a lot
